src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -4,17 +4,27 @@
 import java.util.Map;
 
 public class FruitDaoImpl implements FruitDao {
+    private final Map<String, Integer> storage;
+
+    public FruitDaoImpl() {
+        this.storage = FruitsStorage.fruitsStorage;
+    }
+
+    public FruitDaoImpl(Map<String, Integer> storage) {
+        this.storage = storage;
+    }
+
     @Override
     public Map<String, Integer> getData() {
-        if (FruitsStorage.fruitsStorage.isEmpty()) {
+        if (storage.isEmpty()) {
             throw new RuntimeException("Fruit storage is empty!");
         }
-        return FruitsStorage.fruitsStorage;
+        return storage;
```

Comment:
why are you throwing an exception in the case of empty data? maybe it should just return empty list. but it's just my opinion, btw.